pkg/apis/monitoring/v1/types.go

```diff
@@ -249,6 +249,24 @@ type PrometheusSpec struct {
 	Thanos *ThanosSpec `json:"thanos,omitempty"`
 	// Priority class assigned to the Pods
 	PriorityClassName string `json:"priorityClassName,omitempty"`
+	// FilesystemAccessThroughServiceMonitors configures whether configuration
+	// based on a service monitor can access assets on the file system of the
+	// Prometheus container e.g. bearer tokens.
+	FilesystemAccessThroughServiceMonitors FilesystemAccessThroughServiceMonitorsConfig `json:"filesystemAccessThroughServiceMonitors"`
```

Comment:
Naming for `FilesystemAccessThroughServiceMonitors` is up for discussion. This is merely to make the pull request itself easier to understand.